### ISSUE TITLE
no space outside of table causes it to always float (cover first rows)

### ISSUE THREAD

**Issue by fiendish** [mkoryak/floatThead#180]
If there are no pixels (body margin:0px, padding:0px, etc) between the top of the page and the table, setting the table to windowscroll always triggers the floating condition which means the first row(s) are always partially hidden.

I fixed this for my case (window scroll, absolute) by simply changing the > to >= in line
https://github.com/mkoryak/floatThead/blob/5e5f08069835fb214de8174d4e213d8a4e2b0f8b/dist/jquery.floatThead.js#L618

I'm going to guess that the arithmetic is similar for other scrollings.

**Comment by mkoryak** [comment]
Thanks for finding the fix. Do you mind throwing up a jsfiddle that demonstrates this bug so I can have something to test against?

**Comment by fiendish** [comment]
http://jsfiddle.net/9eo8gjag/

Change the body css from margin 0 to margin 1.

There exists a second obvious problem in the posted fiddle, however. Even after changing the margin to 1, scrolling at all completely breaks the arrangement.

**Comment by mkoryak** [comment]
I see your clever trick to get around the error you would get when you run this plugin on a table which does not have a thead. this is what causes it to fail on reflow, see update:

http://jsfiddle.net/9eo8gjag/1/

I see the bug that you outlined, thanks. Ill have a fix soon

**Comment by fiendish** [comment]
I had a thead, but you are right I accidentally did not have a tbody tag! (oops. that is a mistake that is not present in my project though).
Updated.

http://jsfiddle.net/9eo8gjag/2/

Is it thus necessary to have a tbody defined before calling floatThead? 
Compare 
http://jsfiddle.net/9eo8gjag/3/

**Comment by fiendish** [comment]
Oops. Actually update 2 is not what I meant to show. I'm doing a terrible job of cutting down the one I have from before...hold on...

**Comment by fiendish** [comment]
Here. (finally, I think?)
Compare http://jsfiddle.net/9eo8gjag/4/ with http://jsfiddle.net/9eo8gjag/3/

**Comment by mkoryak** [comment]
yep, that looks like a bug to me. Ill get a fix in for it soon

**Comment by fiendish** [comment]
Reflow after adding tbody doesn't seem to help.
http://jsfiddle.net/9eo8gjag/5/

**Comment by mkoryak** [comment]
nice, i see a different bug here that ive been trying to fix for a while but didnt have a good way to reproduce:  http://jsfiddle.net/9eo8gjag/6/

thanks for the reports.

**Comment by fiendish** [comment]
Sorry if this is abusing the issue queue, but is your other bug the different scrolling behavior where the tbody jumps when you start scrolling if it's not created before the first floatThead and scrolls smoothly if declared before?
( scrolling behavior http://jsfiddle.net/9eo8gjag/7/  vs  http://jsfiddle.net/9eo8gjag/6/ )
I can make a new issue if you want. I know the thread is diverging.

**Comment by mkoryak** [comment]
my other bug is that floatThead does not init itself properly after you add the body. remove the plugin code and you will see that the table should be much wider after you stick that html into it.

**Comment by mkoryak** [comment]
Heh, you are in sommerville, I am in JP. small world :)

your original issue should be fixed on master. I made the change you suggested.
 
I just noticed the scrolling/jumping bug you pointed out. I think you are the first person add a tbody after plugin init, but I think its a valid usecase and should work. If you want to make a new issue that would be great.

I will probably make a 3rd issue for myself with the size thing, but that one is going to be hard to fix.
